### ISSUE TITLE
Opening hydrostar mesh

### ISSUE THREAD

**Issue by martijnwilpshaar** [mancellin/capytaine#27]
Hi,

When opening a hydrostar (.hst) mesh, it gives the following error: 

File "C:\Anaconda3\lib\site-packages\capytaine\io\mesh_loaders.py", line 151, in load_HST
    vertices_tmp.append(list(map(float, node.split()[1:])))

AttributeError: 'numpy.ndarray' object has no attribute 'append'

I'm interested in opening this .hst mesh and see the underwater volume (which is necessary for my further calculations). This mesh has been generated by an external company so I cannot share the details unfortunately. 

Could you have a look?

Many thanks!

**Comment by mancellin** [comment]
Yes, there seems to be an issue with the mesh loader.
I'm not familiar with the HST format, so it won't be easy to fix without an example of a failing file.

@frongere Do you remember the logic behind [this section of the code?](https://github.com/LHEEA/meshmagick/blob/37c5b1c0eaa360ebe1ac161022986f46c87b073b/meshmagick/mmio.py#L148)

**Comment by dirkpir** [comment]
I am also having trouble opening a Hydrostar mesh although I get a different error message:

_AssertionError: Faces of a mesh should be provided as positive integers (ids of vertices)_

Replacing the following line of code in **def load_HST**

_return Mesh(vertices, faces-1, name)_

by

_return Mesh(vertices, faces, name)_

fixed this issue when loading the hydrostar mesh (as by taking faces-1 results in a face being represented by a negative integer -1)

However, when trying to solve the diffraction/radiation problem, the GRMS solver did not converge, and the direct linear solver resulted in an empty dataset.

I attached the hydrostar mesh (replacing the .hst extension by .txt to be able to attach it)

[Ship.txt](https://github.com/mancellin/capytaine/files/6998793/Ship.txt)

**Comment by mancellin** [comment]
Indeed, the mesh reader does not seem to be able to read all possible variants of HST files. There are several issues here:

* as you noticed, the mesh reader expects the vertices to be numbered starting from 1, and not from 0. The Hydrostar documentation does not seem to specify how the vertices should be numbered (but the examples there start at 1).
* your panels are defined as triangles of the form
  ```
  panel_index vertex_1 vertex_2 vertex_3
  ```
  but are wrongly interpreted as a list of quadrangles:
  ```
  vertex_1 vertex_2 vertex_3 vertex_4
  ```
  The mesh reader is at fault here.
* Finally, the symmetry information in the file are ignored by the mesh reader (but could be since Capytaine has `SymmetricMesh` types, unlike meshmagick for which this code was written in the first place.)

**Comment by mancellin** [comment]
Since I'm looking at the HST file reader:
I believe @martijnwilpshaar's issue was with the support of multi-body files. It seems that the mesh reader was thought to support several blocks of vertices coordinates, but the Python code is clearly wrong. If anybody has an example of a multibody Hydrostar mesh, it would help to debug this.

**Comment by dirkpir** [comment]
@mancellin thanks for your very rapid response!
I managed to load in the mesh by replacing panel_index by one of the vertices in the **load_HST** code.

I now mirrored the body to construct a mesh of the full body (ignoring symmetries). Is there any way to tell the solver that this mesh is symmetric to exploit that and speed up the computation?

**Comment by mancellin** [comment]
> I now mirrored the body to construct a mesh of the full body (ignoring symmetries). Is there any way to tell the solver that this mesh is symmetric to exploit that and speed up the computation?

See this example from the cookbook:

https://github.com/mancellin/capytaine/blob/2d1be0434b6343808ac63ca810f9dce8a032ebd5/docs/user_manual/examples/symmetric_body.py#L19-L21

**Comment by dirkpir** [comment]
> > I now mirrored the body to construct a mesh of the full body (ignoring symmetries). Is there any way to tell the solver that this mesh is symmetric to exploit that and speed up the computation?
> 
> See this example from the cookbook:
> 
> https://github.com/mancellin/capytaine/blob/2d1be0434b6343808ac63ca810f9dce8a032ebd5/docs/user_manual/examples/symmetric_body.py#L19-L21

Thanks!
